### PR TITLE
Add command (`/marks`) slack integration

### PR DESCRIPTION
**Pull request by anaelChardan**

### CODE REVIEW

File: src/AppBundle/Controller/SlackController.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+
+class SlackController extends Controller
+{
+    /**
+     * @Route("/slacks", name="slack end point")
```
Comment (anaelChardan) [inline src/AppBundle/Controller/SlackController.php]: no spaces in name :)

File: src/AppBundle/Controller/SlackController.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+     */
+    public function indexAction(Request $request)
+    {
+        $logger = $this->get('logger');
+        $slack_service = $this->get("marks_crawler.core.slack");
```
Comment (anaelChardan) [inline src/AppBundle/Controller/SlackController.php]: camlCase

File: src/AppBundle/Controller/SlackController.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+
+        $logger->info("Received request");
+        $logger->info($request);
+
+        $response_url = $request->request->get('response_url');
```
Comment (anaelChardan) [inline src/AppBundle/Controller/SlackController.php]: $request->get('response_url') is enough

File: src/AppBundle/Controller/SlackController.php [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        $logger->info('$response_url = ' . $response_url);
+
+        $user_id = $request->request->get('user_id');
```
Comment (anaelChardan) [inline src/AppBundle/Controller/SlackController.php]: $request->get('user_id')

File: src/AppBundle/Controller/SlackController.php [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        $logger->info('$user_id = ' . $user_id);
```
Comment (anaelChardan) [inline src/AppBundle/Controller/SlackController.php]: "user_id = {$user_id}" or sprintf('user id = %s', $user_id)

File: src/Core/Mailer/Mailer.php [hunk 6]
```diff
@@ -25,19 +25,22 @@ public function __construct(\Swift_Mailer $mailer, TwigEngine $templating)
...
-            ->setTo($user->getMail())
+            ->setTo($email)
             ->setBody(
                 $this->templating->render(
                     'Emails/report.html.twig',
                     [
-                        'modules' => array_filter($user->getModules(), function(Module $module) {
-                            return !empty($module->getMarks());
-                        })
+                        'modules' => $report['courses']
```
Comment (anaelChardan) [inline src/Core/Mailer/Mailer.php]: Where is report ? I think you have not rebase yet ?

File: src/Core/Slack/Slack.php [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+        $this->mailer = $mailer;
+    }
+
+
+    public function sendResponseToSlack($url, $user_id){
+        $user_email = $this->user->getEmail($user_id);
+
+        $client = new Client();
+
+        $client->request('POST', $url, [
```
Comment (anaelChardan) [inline src/Core/Slack/Slack.php]: $client->post if you want :)

File: src/Core/User/User.php [hunk 8]
```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Core\User;
+
+use Symfony\Bundle\TwigBundle\TwigEngine;
+use GuzzleHttp\Client;
+
+class User
```
Comment (anaelChardan) [inline src/Core/User/User.php]: You could put it in the already existing object User

File: src/Core/User/User.php [hunk 9]
```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Core\User;
+
+use Symfony\Bundle\TwigBundle\TwigEngine;
```
Comment (anaelChardan) [inline src/Core/User/User.php]: ???????

File: src/Core/User/User.php [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+
+    private $slack_api_token;
+
+    public function __construct($slack_api_token)
+    {
+        $this->slack_api_token = $slack_api_token;
+
+    }
+
+    public function getUserInfo($user_id)
```
Comment (anaelChardan) [inline src/Core/User/User.php]: Should not be on object user but on slack service

File: src/AppBundle/Controller/SlackController.php [hunk 11]
```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace AppBundle\Controller;
+
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+
+class SlackController extends Controller
```
Comment (anaelChardan) [inline src/AppBundle/Controller/SlackController.php]: You could do a controller as a service which is better :) (see on doc)

File: src/Core/User/User.php (hunk 8 above)
Comment (Dnomyar) [inline src/Core/User/User.php]: Yes but AppBundle\Model\User is a model and this class is a service. It should be separed, should not be ?

File: src/AppBundle/Controller/SlackController.php (hunk 1 above)
Comment (Dnomyar) [inline src/AppBundle/Controller/SlackController.php]: Underscores ?

Comment (anaelChardan) [inline src/AppBundle/Controller/SlackController.php]: Yup

File: src/Core/User/User.php (hunk 8 above)
Comment (anaelChardan) [inline src/Core/User/User.php]: Yup, But you can seperated your features, indead you can store the token on the user object and have a service decorator to inspect it onto slack, but imho you can have the getUserInfo into the slack service which is the entrypoint for dialog to slack.

**Pull request merged**